### ISSUE TITLE
Rule Request: Disallow Result.get()

### ISSUE THREAD

**Issue by 72A12F4E** [realm/SwiftLint#2634]
### New rule request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

With Swift 5's new Result type people will inevitably try to 'unwrap' them using `.get()`, when a simple `.map` or `.flatMap` would suffice for their needs. `do/catch` already introduces additional scoping, and really doesn't provide any advantages over mapping. 

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

**Bad**

```swift
let integerResult: Result<Int, Error> = .success(5)
let value = try! integerResult.get()
```

```swift
let integerResult: Result<Int, Error> = .success(5)
do {
    let value = try integerResult.get()
    ///etc...
} catch {
    ///etc..
}
```

**Ok**

```swift
let notAResult = Foo()
notAResult.get()
```

```swift
let integerResult: Result<Int, Error> = .success(5)
integerResult.map {
    $0 + 1
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

    No configuration needed.

4. Should the rule be opt-in or enabled by default? Why?

    This should definitely be 100% opt-in. the `do/catch` methodology is a valid one. I just think using `.map` or `.flatMap` is better practice.
